src/cpp/fastdds/xtypes/utils.cpp

```diff
@@ -0,0 +1,1031 @@
+    {
+        output << j;
+    }
+    else
+    {
+        EPROSIMA_LOG_WARNING(XTYPES_UTILS, "Error encountered while performing DynamicData to JSON serialization.");
+    }
+    return ret;
+}
+
+ReturnCode_t json_serialize(
+        const DynamicData::_ref_type& data,
+        std::string& output,
+        DynamicDataJsonFormat format) noexcept
+{
+    ReturnCode_t ret;
+    std::stringstream ss;
+
+    if (RETCODE_OK == (ret = json_serialize(data, ss, format)))
+    {
+        output = ss.str();
+    }
+    else
+    {
+        EPROSIMA_LOG_WARNING(XTYPES_UTILS, "Error encountered while performing DynamicData to JSON serialization.");
+    }
+    return ret;
+}
+
+ReturnCode_t json_serialize(
+        const traits<DynamicDataImpl>::ref_type& data,
+        nlohmann::json& output,
+        DynamicDataJsonFormat format) noexcept
+{
+    if (nullptr != data)
+    {
+        switch (data->type()->get_kind())
+        {
+            case TK_STRUCTURE:
+            {
+                ReturnCode_t ret = RETCODE_OK;
+                DynamicTypeMembersById members;
+                if (RETCODE_OK != (ret = data->type()->get_all_members(members)))
+                {
+                    EPROSIMA_LOG_WARNING(XTYPES_UTILS,
+                            "Error encountered while serializing structure to JSON: get_all_members failed.");
+                    return ret;
+                }
+                for (auto it : members)
+                {
+                    if (RETCODE_OK != (ret = json_serialize_member(data, it.second, output, format)))
+                    {
+                        EPROSIMA_LOG_WARNING(XTYPES_UTILS,
+                                "Error encountered while serializing structure member to JSON.");
+                        break;
+                    }
+                }
+                return ret;
+            }
+            default:
+            {
+                EPROSIMA_LOG_WARNING(XTYPES_UTILS, "Only structs are supported by json_serialize method.");
+                return RETCODE_BAD_PARAMETER;
+            }
+        }
+    }
+    else
+    {
+        EPROSIMA_LOG_WARNING(XTYPES_UTILS,
+                "Encountered null data value while performing DynamicData to JSON serialization.");
+        return RETCODE_BAD_PARAMETER;
+    }
+}
+
+ReturnCode_t json_serialize_member(
+        const traits<DynamicDataImpl>::ref_type& data,
+        const traits<DynamicTypeMember>::ref_type& type_member,
+        nlohmann::json& output,
+        DynamicDataJsonFormat format) noexcept
+{
+    MemberDescriptorImpl& member_desc =
+            traits<DynamicTypeMember>::narrow<DynamicTypeMemberImpl>(type_member)->get_descriptor();
+
+    return json_serialize_member(data, type_member->get_id(),
+                   traits<DynamicType>::narrow<DynamicTypeImpl>(
+                       member_desc.type())->resolve_alias_enclosed_type()->get_kind(),
```

Comment:
Extend DynamicTypeImpl to have a new attribute + getter for its enclosing type, so the recursive search is only done once.

Reply:
Hello @richiware , any thoughts on this?

Reply:
Taking a note...

Reply:
We decided not to implement this enhancement for the moment.